test/manual/tls_support.test.ts

```diff
@@ -42,41 +43,79 @@ describe('TLS Support', function () {
 
   context('when tls filepaths are provided', () => {
     let client: MongoClient;
+
     afterEach(async () => {
-      if (client) await client.close();
+      await client?.close();
     });
 
     context('when tls filepaths have length > 0', () => {
-      beforeEach(async () => {
-        client = new MongoClient(CONNECTION_STRING, tlsSettings);
-      });
+      context('when connection will succeed', () => {
+        beforeEach(async () => {
+          client = new MongoClient(CONNECTION_STRING, tlsSettings);
+        });
+
+        it('should read in files async at connect time', async () => {
+          expect(client.options).property('tlsCAFile', TLS_CA_FILE);
+          expect(client.options).property('tlsCertificateKeyFile', TLS_CERT_KEY_FILE);
+          expect(client.options).not.have.property('ca');
+          expect(client.options).not.have.property('key');
+          expect(client.options).not.have.property('cert');
+
+          await client.connect();
+
+          expect(client.options).property('ca').to.exist;
+          expect(client.options).property('key').to.exist;
+          expect(client.options).property('cert').to.exist;
+        });
+
+        context('when client has been opened and closed more than once', function () {
+          it('should only read files once', async () => {
+            await client.connect();
+            await client.close();
 
-      it('should read in files async at connect time', async () => {
-        expect(client.options).property('tlsCAFile', TLS_CA_FILE);
-        expect(client.options).property('tlsCertificateKeyFile', TLS_CERT_KEY_FILE);
-        expect(client.options).not.have.property('ca');
-        expect(client.options).not.have.property('key');
-        expect(client.options).not.have.property('cert');
+            const caFileAccessTime = (await fs.stat(TLS_CA_FILE)).atime;
+            const certKeyFileAccessTime = (await fs.stat(TLS_CERT_KEY_FILE)).atime;
 
-        await client.connect();
+            await client.connect();
 
-        expect(client.options).property('ca').to.exist;
-        expect(client.options).property('key').to.exist;
-        expect(client.options).property('cert').to.exist;
+            expect((await fs.stat(TLS_CA_FILE)).atime).to.deep.equal(caFileAccessTime);
+            expect((await fs.stat(TLS_CERT_KEY_FILE)).atime).to.deep.equal(certKeyFileAccessTime);
+          });
+        });
       });
 
-      context('when client has been opened and closed more than once', function () {
-        it('should only read files once', async () => {
-          await client.connect();
-          await client.close();
+      context('when the connection will fail', () => {
```

Comment:
What do these tests mean?  I interpret them as "when the TLS settings are misconfigured" (resulting in the call to `connect` to fail.  And the first test does demonstrate this.

But what about the second?  I'd expect the call to `connect` on line 109 to throw, if the client were misconfigured, and that the test would fail.  How is it passing?

Reply:
The connection will fail here because the CRL will cause a revokation. The second was failing - I just fixed that test.